src/simplify_qdq.cpp

```diff
@@ -0,0 +1,166 @@
+            match::arg(2)(match::skip_broadcasts(match::all_of(match::has_value(0)))));
+    }
+
+    auto matcher() const
+    {
+        return match::name(get_quantizable_op_names())(
+            match::arg(0)(dequantizelinear_op("x1", "scale1")),
+            match::arg(1)(dequantizelinear_op("x2", "scale2")));
+    }
+
+    void apply(module& m, match::matcher_result r) const
+    {
+        auto qop    = r.result;
+        auto q1     = r.instructions["x1"];
+        auto q2     = r.instructions["x2"];
+        auto scale1 = r.instructions["scale1"];
+        auto scale2 = r.instructions["scale2"];
+
+        // Only INT8 type currently supported
+        if(q1->get_shape().type() != migraphx::shape::int8_type or
+           q2->get_shape().type() != migraphx::shape::int8_type)
+            return;
+
+        double scale;
+        visit_all(scale1->get_literal(), scale2->get_literal())(
+            [&](const auto s1, const auto s2) { scale = s1.front() * s2.front(); });
+
+        auto qop_args  = qop->inputs();
+        qop_args.at(0) = q1;
+        qop_args.at(1) = q2;
+        instruction_ref dq;
+        instruction_ref dq_scale;
+        instruction_ref zero_point;
+        if(qop->name() == "convolution")
+        {
+            auto conv_val = qop->get_operator().to_value();
+            dq            = m.insert_instruction(
+                qop, migraphx::make_op("quant_convolution", conv_val), qop_args);
+        }
+        else if(qop->name() == "dot")
+        {
+            auto dot_op = any_cast<op::dot>(qop->get_operator());
+            if(!(float_equal(dot_op.alpha, 1.0f) and float_equal(dot_op.beta, 0.0f)))
+                return;
+            if(qop_args.size() == 3)
+                qop_args.pop_back();
+            dq = m.insert_instruction(
+                qop, migraphx::make_op("quant_dot", {{"alpha", 1}, {"beta", 0}}), qop_args);
+        }
+        dq_scale   = m.add_literal(static_cast<float>(scale));
+        zero_point = m.add_literal(0);
+
+        auto lens = dq->get_shape().lens();
+        auto scale_mb =
+            m.insert_instruction(qop, make_op("multibroadcast", {{"output_lens", lens}}), dq_scale);
+        auto shift_mb = m.insert_instruction(
+            qop, make_op("multibroadcast", {{"output_lens", lens}}), zero_point);
+        dq = m.insert_instruction(qop, make_op("dequantizelinear"), dq, scale_mb, shift_mb);
+        m.replace_instruction(qop, dq);
+    }
+};
+
+bool compare_literals(instruction_ref ins1, instruction_ref ins2)
+{
+    if(ins1->name() == "broadcast" or ins1->name() == "multibroadcast")
+        ins1 = ins1->inputs().front();
+    if(!ins1->can_eval())
+        return false;
+    auto literal1 = ins1->get_literal();
+
+    if(ins2->name() == "broadcast" or ins2->name() == "multibroadcast")
+        ins2 = ins2->inputs().front();
+    if(!ins2->can_eval())
+        return false;
+    auto literal2 = ins2->get_literal();
+
+    bool are_equal = false;
```

Comment:
There is another scenario that: if literal1 and literal2 are of different shapes, but all their values are the same, this function will return false, while it should return true. So the comparison can be written as:

```
    bool are_equal = false;
    visit_all(literal1, literal2)([&](const auto l1, const auto l2) {
        bool equal1 = std::equal(l1.begin(), l1.end(), l2.begin(), float_equal);
        auto val = *l1.begin();
        bool equal2 = std::all_of(l1.begin() + 1, l1.end(), [&](auto v) { return float_equal(v, val); }) and 
             std::all_of(l2.begin(), l2.end(), [&](auto v) { return float_equal(v, val); });
        are_equal = equal1 or equal2;
    });
```